### PR TITLE
Refactored queue to use `rabbitmq-client`

### PR DESCRIPTION
**Pull request by iboshkov**
## What has been done?

 - [x] Reimplemented queueing to use https://www.npmjs.com/package/rabbitmq-client
 - [x] Added CPU limit setup on executioner
 - [x] Updated Strapi to 4.21 
 - [x] Fixed a bug where the RabbitMQ subscription would stay open during `strapi develop`

### CODE REVIEW

File: executioner/languages/docker.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ type File = {
   metadata?: any;
 };
 
-const DEFAULT_EXECUTION_TIMEOUT = parseInt(process.env.DEFAULT_EXECUTION_TIMEOUT || '60');
+const CPU_LIMIT_PER_EXECUTION = process.env.CPU_LIMIT_PER_EXECUTION;
```
Comment (gjorgi-krenkov) [inline executioner/languages/docker.ts:21]: Can you name this as MAX_CPU_LIMIT_PER_EXECUTION, as the CP_LIMIT_PER_EXECUTION will come from the problem in the future

File: senatus/src/index.ts [hunk 2]
```diff
@@ -20,34 +23,49 @@ export default {
...
+          const existing = await strapi.entityService.findOne('api::execution.execution', result.metadata.executionId, {
+            populate: ['testCase'],
+          });
+          const execution = await strapi.entityService.update('api::execution.execution', result.metadata.executionId, {
+            data: {
+              stdout: result.stdout,
+              stderr: result.stderr,
+              executionTime: result.time,
+              processed: true,
+              passed: existing.testCase.output === result.stdout,
```
Comment (gjorgi-krenkov) [inline senatus/src/index.ts:54]: I think there was a small issue with stdout adding extra newline at the end, maybe if it's a small thing you could remove it now

**Review by gjorgi-krenkov — APPROVED** [review]

File: senatus/src/index.ts (hunk 2 above)
Comment (iboshkov) [reply senatus/src/index.ts:54]: That's due to the user-provided program, so we would need to align the expected output in this case, I think trimming the output would be a mistake.

File: executioner/languages/docker.ts (hunk 1 above)
Comment (iboshkov) [reply executioner/languages/docker.ts:21]: We need to do some rework on how we execute and how we do isolation per test case execution

**Review by vergjor — APPROVED** [review]

**Comment by vergjor** [comment]
@iboshkov I think we can merge this now right? 😄

**Pull request merged**